client/src/main/java/com/vesoft/nebula/client/graph/net/SyncConnection.java

```diff
@@ -102,16 +108,19 @@ public void open(HostAddress address, int timeout, SSLParam sslParam, boolean is
     @Override
     public void open(HostAddress address, int timeout) throws IOErrorException,
             ClientServerIncompatibleException {
-        this.open(address, timeout, false);
+        this.open(address, timeout, false, headers);
     }
 
     @Override
-    public void open(HostAddress address, int timeout, boolean isUseHttp2)
+    public void open(HostAddress address, int timeout,
+                     boolean isUseHttp2, Map<String,String> headers)
             throws IOErrorException, ClientServerIncompatibleException {
         try {
             this.serverAddr = address;
             this.timeout = timeout <= 0 ? Integer.MAX_VALUE : timeout;
-            if (isUseHttp2) {
+            this.useHttp2 = isUseHttp2;
+            this.headers = headers;
+            if (useHttp2) {
                 getProtocolForHttp2();
```

Comment:
headers are only for `useHttp2=true`, only used for HTTPClient
prefer `getProtocolForHttp2(headers);` and do not use `this.headers`

Reply:
reopen will use headers when useHttp2 is true, need a global var to save the info.